### DISCUSSION TITLE
Shuffling Playlist

### DISCUSSION THREAD

**Discussion by jordanwmckee** [gilbarbara/react-spotify-web-playback#142] [Q&A] [answered]
I'm working on a project with the player, and want to give users an option to shuffle the playlist that's currently playing. Is there a way to do this with this component, or would I need to create a separate button with some functionality to do so?

**Comment by gilbarbara** [comment, marked as answer]
Hey @jordanwmckee 

There's no way to do this within the player since it uses the `uris` as the source of truth.

I thought about adding extra buttons, like queue and shuffle, in the UI refactor implemented a few weeks ago.
However, that would increase the bundle size and add complications a simple player shouldn't have to handle.

One option would be to accept custom components to use before the previous button or after the next button.
But the previous/next buttons render conditionally depending on the size of `uris` and `offset`, so layout shift could be problematic.

**Comment by gilbarbara** [comment]
Hey @jordanwmckee 
Version 0.14 added support to add extra buttons to the player and export the Spotify API methods used internally.
With that, you can create a Shuffle button. Check the demo.
